lib/Brass/API.pm

```diff
@@ -119,18 +119,19 @@ get 'api/pwd/' => sub {
         -cipher => 'Blowfish'
     );
 
-    my $pass;
-    if ($username)
-    {
-        $pass = $cipher->decrypt($username->pwencrypt);
-    }
-    else {
+    my $pass = query_parameters->get('pass');
+    unless ( $pass ) {
+      if ($username)
+        {
+          $pass = $cipher->decrypt($username->pwencrypt);
+        }
+      else {
         my $pw = $cipher->encrypt(randompw);
         my $s = $schema->resultset('Server')->find_or_create({ name => $server });
         my $u = $schema->resultset('Pw')->create({ server_id => $s->id, username => $param, pwencrypt => $pw, type => $action });
         $pass = $cipher->decrypt($u->pwencrypt);
+      }
```

Comment:
I'm not sure if I've misread the above, but am I correct in saying that if ```pass``` is provided as a parameter, it doesn't actually get encrypted and stored in the database, which surely it should?

Reply:
It's encrypted in line 129, and stored in server encrypted in line 131, but returned as plaintext for consistency with the autogenerated passwords.. didn't want to break the UI, etc

Reply:
No.. there's some code missing there... bah!